### DISCUSSION TITLE
point2grid Methods

### DISCUSSION THREAD

**Discussion by cjmelick** [dtcenter/METplus#1768] [Statistical Computation] [answered]
In point2grid, with the methods available, what is probably the best option for gridding categorical data variables (e.g., precipitation type)?   I noticed in the documentation:  Only 4 interpolation methods are applied to the field variables; MIN/MAX/MEDIAN/UW_MEAN.  I was curious is the UW_MEAN a weighted mean based on distance of each observation from the grid point?   And is the radius considered in the interpolation method correspond to the grid point distance?

Thanks,
Chris Melick

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @cjmelick, I see you have a question about the Point2Grid tool in MET.

The processing logic is basically this:
- Loop over the input point observations and filter them by variable type, time, message type, and QC, as needed.
- Convert the (lat, lon) location for each obs into (x, y) grid space, rounding to the NEAREST grid point.
- Loop over the grid points for the target grid and process the point obs which were closest to that point. The output value is computed as their minimum (`-method min`), maximum (`-method max`), median (`-method median`), or average (un-weighted) mean (`-method uw_mean`).

It does NOT currently support the computation of a distance-weighted mean. Logically, that could be done but adding it isn't trivial based on the way the logic is currently implemented.

The point2grid tool is interesting in that we need to consider issues at the opposite ends of a spectrum. When processing point observation station data or storm reports, in general, the data is so sparse it's difficult to create a meaningful result. Conversely, when processing very dense GOES (lat, lon) pixels, the data is so dense, processing can be slow.

The density (or scarcity) of input obs is something you should think about. For sparse input, use a coarser grid so that *enough* of them fall within each grid box to get a good result. Although, I definitely see that distance-weighting (`-method dw_mean`) would be really useful.

As for categorical variables, like PTYPE, none of the existing options make sense. Using the nearest neighbor (`-method nearest`) would probably be best. But that also is not currently supported. This is still a relatively new tool on which we haven't received much feedback. Do you see a clear need to support the nearest neighbor and distance weighting interpolation methods?

**Reply by JohnHalleyGotway** [reply]
FYI, I wrote this up as a new GitHub issue: https://github.com/dtcenter/MET/issues/2245

Please feel free to comment on that issue to clarify details about need, intended use, and timelines.

**Reply by cjmelick** [reply]
Thanks @JohnHalleyGotway!   I think adding on those two method types based on distance would be a great enhancement to MET point2grid moving forward.   Another possibility with categorical type data is maybe detecting which value occurs the most near a grid point.   I guess that would be more like a MODE statistic (and by extension even a distance weighted MODE).   I am sure many end users would see benefit in these added options to creating gridded analyses.   

Thanks again,
Chris

**Reply by JohnHalleyGotway** [reply]
Ah yes, good point, I added a note about supporting `-method mode` to the dtcenter/MET#2245 issue.

And actually, that could be a good interpolation method to support more broadly across the MET tools. It might be useful when regridding categorical data, in particular.

**Reply by cjmelick** [reply]
Hey John,

I did think of something to incorporate if going down the path of better capturing artifacts of gridding categorical variable observations.   It would be advantageous to give counts for each category value, kind of like you are doing already for specifying the number of records nearby for the variable for each grid point.   I guess in that case there might be a flag to switch on and off just for this purpose.  

Thanks,
Chris

**Reply by JohnHalleyGotway** [reply]
Chris, just cleaning up some old discussions. I added your feedback as a [comment](https://github.com/dtcenter/MET/issues/2245#issuecomment-1259771842) on that issue. I'll go ahead and mark this discussion as being answered and lock it to prevent future posts. Please feel free to comment on that issue with more details as needed.
